target/linux/ramips/modules.mk

```diff
@@ -72,18 +72,58 @@ endef
 
 $(eval $(call KernelPackage,i2c-mt7621))
 
+define KernelPackage/dma-ralink
+  SUBMENU:=Other modules
+  TITLE:=Ralink GDMA Engine
+  DEPENDS:=@TARGET_ramips
+  KCONFIG:= \
+	CONFIG_DMADEVICES=y \
+	CONFIG_DW_DMAC_PCI=n \
+	CONFIG_DMA_RALINK
+  FILES:= \
+	$(LINUX_DIR)/drivers/dma/virt-dma.ko \
+	$(LINUX_DIR)/drivers/dma/ralink-gdma.ko
+  AUTOLOAD:=$(call AutoLoad,52,ralink-gdma)
+endef
+
+define KernelPackage/dma-ralink/description
+ Kernel modules for enable ralink dma engine.
+endef
+
+$(eval $(call KernelPackage,dma-ralink))
+
+define KernelPackage/hsdma-mtk
+  SUBMENU:=Other modules
+  TITLE:=MediaTek HSDMA Engine
+  DEPENDS:=@TARGET_ramips @TARGET_ramips_mt7621
+  KCONFIG:= \
+	CONFIG_DMADEVICES=y \
+	CONFIG_DW_DMAC_PCI=n \
+	CONFIG_MTK_HSDMA
+  FILES:= \
+	$(LINUX_DIR)/drivers/dma/virt-dma.ko \
+	$(LINUX_DIR)/drivers/dma/mtk-hsdma.ko
+  AUTOLOAD:=$(call AutoLoad,53,mtk-hsdma)
+endef
+
+define KernelPackage/hsdma-mtk/description
+ Kernel modules for enable MediaTek hsdma engine.
+endef
 
+$(eval $(call KernelPackage,hsdma-mtk))
 
 define KernelPackage/sound-mt7620
   TITLE:=MT7620 PCM/I2S Alsa Driver
-  DEPENDS:=@TARGET_ramips_mt7620 +kmod-sound-soc-core +kmod-regmap @!TARGET_ramips_rt288x
+  DEPENDS:=@TARGET_ramips +kmod-sound-soc-core +kmod-regmap +kmod-dma-ralink @!TARGET_ramips_rt288x
   KCONFIG:= \
```

Comment:
mt7621 has gdma and hsdma. gdma for slow devices like i2s. hsdma only support mem to mem copy. so mt7621 don't need hsdma for audio.

the gdma and i2s functions support on all ramips/mediatek targets except rt288x.
so i remote the mt7620 limit. because the mt7628 and mt7621 should support audio too.